File: algo_problems/q641-660/q645.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        int[] flags = new int[nums.length + 1];
        flags[0] = 1;
        for (int num: nums) {
        	flags[num] += 1;
        }
        int[] result = new int[2];
        result[0] = Array.asList(flags).indexOf(2);
        result[1] = Array.asList(flags).indexOf(0);
        return result;
    }
}

File: algo_problems/q641-660/q645.py
from collections import Counter


class Solution:
    def findErrorNums(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        n = len(nums)
        c = Counter(nums)
        lack = -1
        duplicated = -1
        for i in range(1, n + 1):
            if c[i] == 0:
                lack = i
            elif c[i] == 2:
                duplicated = i
            if lack != -1 and duplicated != -1:
                break
        return [duplicated, lack]